tycho-indexer/src/extractor/dynamic_contract_indexer/hook_dci.rs

```diff
@@ -2005,8 +2011,10 @@ mod tests {
             // Use real RPC-based tracer instead of mock
             let trace_rpc_url = std::env::var("TRACE_RPC_URL").expect("RPC_URL must be set");
             let rpc_url = std::env::var("RPC_URL").expect("RPC_URL must be set");
-            let entrypoint_tracer = EVMEntrypointService::try_from_url(&trace_rpc_url)
-                .expect("Failed to create RPC entrypoint tracer");
+            let rpc =
+                EthereumRpcClient::new(&rpc_url).expect("Failed to create Ethereum RPC client");
```

Comment:
Here, we in-fact have a dependency to use 2 different RPCs, TRACE_RPC and RPC_URL.
This happens because the current available methods are not available on all Ethereum RPCs, so we need to have one RPC that fulfills the RPC_URL requirements and another for TRACE_RPC.

So, for the EVMEntrypointService, the correct would be to use trace_rpc_url, like before, and create another client that will be used for setup_test_metadata_registries and setup_test_hook_orchestrator_registry.